backend/de.metas.business/src/main/java/de/metas/invoice/service/impl/CreditMemoInvoiceCopyHandler.java

```diff
@@ -114,178 +110,24 @@ private void handlePartialRequests(final de.metas.adempiere.model.I_C_Invoice in
-
-		final List<de.metas.adempiere.model.I_C_InvoiceLine> lines = invoicePA.retrieveLines(creditMemo, trxName);
-		for (final I_C_InvoiceLine creditMemoLine : lines)
-		{
-			if (creditMemoLine.isDescription())
-			{
-				continue; // not changing description lines
-			}
-			final I_C_InvoiceTax invoiceTax = taxId2Invoicetax.get(creditMemoLine.getC_Tax_ID());
-			List<I_C_InvoiceLine> linesForInvoiceTax = tax2lines.get(invoiceTax);
-			if (linesForInvoiceTax == null)
-			{
-				linesForInvoiceTax = new ArrayList<>();
-				tax2lines.put(invoiceTax, linesForInvoiceTax);
-
-			}
-			linesForInvoiceTax.add(creditMemoLine);
-		}
-
-		final Map<I_C_InvoiceLine, BigDecimal> line2newLineGrossAmt = new IdentityHashMap<>();
-		for (final I_C_InvoiceTax invoiceTax : invoiceTaxes)
-		{
-			BigDecimal sumPerTax = BigDecimal.ZERO;
-			for (final I_C_InvoiceLine creditMemoLine : tax2lines.get(invoiceTax))
-			{
-				final BigDecimal newLineNetAmt = creditMemoLine.getLineNetAmt().multiply(openfraction);
-				final BigDecimal newLineGrossAmt;
-				if (invoiceTax.isTaxIncluded())
-				{
-					newLineGrossAmt = newLineNetAmt.setScale(precision.toInt(), RoundingMode.FLOOR);
-				}
-				else
-				{
-					newLineGrossAmt = newLineNetAmt
-							.multiply(invoiceTax.getC_Tax().getRate()
-									.add(Env.ONEHUNDRED)
-									.divide(Env.ONEHUNDRED))
-							.setScale(precision.toInt(), RoundingMode.FLOOR);
-				}
-				line2newLineGrossAmt.put(creditMemoLine, newLineGrossAmt);
-				sumPerTax = sumPerTax.add(newLineGrossAmt);
-			}
-
-			for (final I_C_InvoiceLine creditMemoLine : tax2lines.get(invoiceTax))
-			{
-				final BigDecimal targetSum = newTaxAmounts.get(invoiceTax);
-				if (sumPerTax.compareTo(targetSum) == 0)
-				{
-					break;
-				}
-				if (sumPerTax.compareTo(targetSum) < 0)
-				{
-					line2newLineGrossAmt.put(creditMemoLine, line2newLineGrossAmt.get(creditMemoLine).add(smallestAmtInCurrency));
-					sumPerTax = sumPerTax.add(smallestAmtInCurrency);
-				}
-				else
-				{
-					Check.errorIf(true, "invoiceTax {} has has targetSum={}, but the credit memo lines for this tax have sumPerTax={} (difference may not be more than {})",
-							invoiceTax, targetSum, sumPerTax, smallestAmtInCurrency);
-				}
-			}
-		}
-
-		for (final I_C_InvoiceLine creditMemoLine : lines)
+		else
 		{
-			creditMemoLine.setQtyEntered(ONE);
-			creditMemoLine.setQtyInvoiced(ONE);
-			creditMemoLine.setPriceEntered(line2newLineGrossAmt.get(creditMemoLine));
-			creditMemoLine.setPriceActual(line2newLineGrossAmt.get(creditMemoLine));
-
-			// 07090: setting priceUOM to 0, because we set the Qty to 1, and the priceUOM only makes sense in the context of the original Qty (but this is a partial-ammount credit memo)
-			InterfaceWrapperHelper.create(creditMemoLine, de.metas.adempiere.model.I_C_InvoiceLine.class).setPrice_UOM_ID(0);
+			logger.warn("The credit memo {} should be a partial allocation of invoice {}.", creditMemo, invoice);
```

Comment:
What the purpose of this warning?
I mean, whom is it for? and what shall the recipient do with it?
I think if we want to make the system more supportable, then debug is enough.
"Warn" messages should imho be actionably

Reply:
I dropped de.metas.invoice.service.impl.CreditMemoInvoiceCopyHandler.handlePartialRequests(I_C_Invoice, I_C_Invoice) as discussed.